src/test/crimson/seastore/test_transaction_manager.cc

```diff
@@ -2173,6 +2174,33 @@ TEST_P(tm_single_device_intergrity_check_test_t, remap_lazy_read)
    });
 }
 
+TEST_P(tm_single_device_test_t, invalid_lba_mapping_detect)
+{
+  run_async([this] {
+    using namespace crimson::os::seastore::lba_manager::btree;
+    {
+      auto t = create_transaction();
+      for (int i = 0; i < LEAF_NODE_CAPACITY; i++) {
+	auto extent = alloc_extent(
+	  t,
+	  i * 4096,
+	  4096,
+	  'a');
+      }
+      submit_transaction(std::move(t));
+    }
+
+    {
+      auto t = create_transaction();
+      auto pin = get_pin(t, (LEAF_NODE_CAPACITY - 1) * 4096);
+      assert(pin->is_parent_valid());
+      auto extent = alloc_extent(t, LEAF_NODE_CAPACITY * 4096, 4096, 'a');
+      assert(!pin->is_parent_valid());
```

Comment:
Probably deserve a further test case when `pin->parent_modified()` becomes true, pin is still useful ?